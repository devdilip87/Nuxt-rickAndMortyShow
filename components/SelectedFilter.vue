<template>
    <section class="row" v-if="selectedFilterOptions.length">
        <h3>Selected Filter</h3>
        <button v-for="(filter, index) in selectedFilterOptions"
          v-bind:key="index"
          @click="removeFilter(filter)">{{ filter.value }}<span class="close-thik"></span></button>
    </section>
</template>

<script>
import { mapGetters } from "vuex";
export default {
  computed: {
      ...mapGetters([
          'selectedFilterOptions'
      ])
  },
  methods: {
      removeFilter: function(filter){
        let removeIndex = this.selectedFilterOptions.map(function(item) { return item.value; }).indexOf(filter.value);
        // remove object
        const selectedFilter = this.selectedFilterOptions.splice(removeIndex, 1);
        const payload = {
            selectedFilter: this.selectedFilterOptions,
            characterList: this.$store.state.characters,
            name: this.$store.state.searchedName
        }
        this.$store.dispatch("updateSelectedFilter", payload);
      }
  }
}
</script>

<style scoped>
    button {
      height      : 25px;
      text-align  : center;
      width       : auto;
      border      : 1px solid;
      padding-left:10px;
      padding-right:10px;
      min-width:80px;
      margin-right: 10px;
      border-radius: 5px;
      background: #2196F3;
      color: white;
      font-size: 14px;
  }

  .close-thik:after {
    padding-left: 10px;
    content: 'âœ–'; /* UTF-8 symbol */
  }

</style>
