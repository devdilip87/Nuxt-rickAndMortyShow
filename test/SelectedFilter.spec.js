import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import SelectedFilter from '@/components/SelectedFilter.vue'

const localVue = createLocalVue();
localVue.use(Vuex);

describe("SelectedFilter view component test case", () => {
    let store;
    let actions;
    let wrapper;
    beforeEach(() => {
        actions = {
            removeFilter: jest.fn()
        };
        store = new Vuex.Store({
        actions
        });

        wrapper = shallowMount(SelectedFilter, {
            data: () => {
                return {
                    selectedFilterOptions: [{type: 2, value: 'Male'}]
                };
            },    
        mocks: {
            $store: {
                getters: {
                    selectedFilterOptions: []
                },
                dispatch: {
                    removeFilter: data => data
                }
            }
        },
        localVue: localVue
        });
    });

    test('generate snapshot testing', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    test("check `selectedFilterOptions` call on computed hooks", () => {
        expect(wrapper.vm.selectedFilterOptions.length).toStrictEqual(1);
    });

    test("check `removeFilter` call on methods", () => {
        wrapper.find('button').trigger('click')
        //expect(wrapper.find('button').text()).toStrictEqual('Male')
        //wrapper.vm.removeFilter({type: 2, value: 'Male'})
        expect(wrapper.vm.selectedFilterOptions).toStrictEqual([])
    });
})