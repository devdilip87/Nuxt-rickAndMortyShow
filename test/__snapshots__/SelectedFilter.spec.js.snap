// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedFilter view component test case generate snapshot testing 1`] = `
"<section class=\\"row\\">
  <h3>Selected Filter</h3> <button>Male<span class=\\"close-thik\\"></span></button>
</section>"
`;
